jxl/src/frame/modular/predict.rs

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) the JPEG XL Project Authors. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+use crate::error::{Error, Result};
+use num_derive::FromPrimitive;
+use num_traits::FromPrimitive;
+
+#[repr(u8)]
+#[derive(Debug, FromPrimitive)]
+pub enum Predictor {
+    Zero = 0,
+    Left = 1,
+    Top = 2,
+    Average0 = 3,
+    Select = 4,
+    Gradient = 5,
+    Weighted = 6,
+    TopRight = 7,
+    TopLeft = 8,
+    LeftLeft = 9,
+    Average1 = 10,
+    Average2 = 11,
+    Average3 = 12,
+    Average4 = 13,
```

Comment:
We are using too many different names for the predictors. The above are from the libjxl code I guess?

predictor index | spec name | jxl_from_tree name | libjxl code | proposal
-- | -- | -- | -- | --
0 | Zero | Set | Zero | Zero
1 | West | W | Left | West
2 | North | N | Top | North
3 | Avg(W,N) | AvgW+N | Average0 | AvgW_N
4 | Select | Select | Select | Select
5 | Gradient | Gradient | Gradient | Gradient
6 | Self-correcting | Weighted | Weighted | Weighted
7 | NorthEast | NE | TopRight | NorthEast
8 | NorthWest | NW | TopLeft | NorthWest
9 | WestWest | WW | LeftLeft | WestWest
10 | Avg(W,NW) | AvgW+NW | Average1 | AvgW_NW
11 | Avg(N,NW) | AvgN+NW | Average2 | AvgN_NW
12 | Avg(N,NE) | AvgN+NE | Average3 | AvgN_NE
13 | AvgAll | AvgAll | Average4 | AvgAll

The rightmost column is what I would propose to use as enum values in both the libjxl and jxl-rs code.

Reply:
Switched to North/East/West names, but somewhat more verbose.

Reply:
That works too, it's not like the length of those enum values really matters anyway...